Sources/PayPalNativeCheckout/PayPalUIInterfaces/NativeCheckout.swift

```diff
@@ -0,0 +1,26 @@
+
+import Foundation
+import UIKit
+@_implementationOnly import PayPalCheckout
+import PaymentsCore
+
+class NativeCheckout: CheckoutProtocol {
+
+    let nxoConfig: CheckoutConfig
+    required internal init(nxoConfig: CheckoutConfig) {
+        self.nxoConfig = nxoConfig
+    }
+
+
+    func start(presentingViewController: UIViewController?,
+                    createOrder: CheckoutConfig.CreateOrderCallback?,
+                    onApprove: CheckoutConfig.ApprovalCallback?,
+                    onShippingChange: CheckoutConfig.ShippingChangeCallback?,
+                    onCancel: CheckoutConfig.CancelCallback?,
+                    onError: CheckoutConfig.ErrorCallback?) {
+        Checkout.set(config: nxoConfig)
+        Checkout.start(presentingViewController: presentingViewController, createOrder: createOrder, onApprove: onApprove, onShippingChange: onShippingChange, onCancel: onCancel, onError: onError)
```

Comment:
Fundamentally this is a good start. There is a small difference [on Android](https://github.com/paypal/Android-SDK/blob/c622da9382bac770b40ed396b48afc06dd30d710/PayPalNativeCheckout/src/main/java/com/paypal/android/checkout/PayPalClient.kt#L67) where we're using a listener to encapsulate the native checkout callback setup. The iOS equivalent here would be a delegate, similar to the one used in the [PayPalWebCheckout](https://github.com/paypal/iOS-SDK/blob/main/Sources/PayPalWebCheckout/PayPalWebCheckoutDelegate.swift) module.

Reply:
this is an internal interface for testing, what you are refering to is on `PayPalClient`

Reply:
Do we need to use a delegate? Is there anything wrong with passing things through closures on both platforms? I'm a bit in the dark about kotlin/java norms so just let me know if that's a silly question. Conforming to a delegate feels like an unnecessary extra step to me, but I've also missed previous context convos on this structure.

Reply:
@jcnoriega ah I see what you mean now I had to double back.

@kaylagalway I will say the path of least resistance for this PR is to keep delegates and create a separate Ticket to refine the developer experience. It'll be good for us to come up with a strategy we can apply to all current (and future) modules that makes sense for the SDK as a whole.

Reply:
Totally agree, that comment was pre-standup convo :)